cinema/views.py

```diff
@@ -136,3 +204,18 @@
 
     def perform_create(self, serializer):
         serializer.save(user=self.request.user)
+
+    def get_permissions(self):
+        if self.action in [
+            "create",
+            "list",
+            "retrieve"
+        ]:
+            permission_classes = [
+                ReadAndCreate,
+            ]
+        else:
+            permission_classes = [
+                NoPermission,
+            ]
+        return [permission() for permission in permission_classes]
```

Comment:
The `get_permissions` method in `OrderViewSet` applies `ReadAndCreate` permission to actions like 'update', 'partial_update', and 'destroy'. Consider using a more restrictive permission for these actions if the intention is to limit them to certain users.